regular_expressions_boundaries_anchors/regular_expressions_boundaries_anchors.md

```diff
@@ -0,0 +1,419 @@
+
+**Acceptable**
+
+* stated, "I want to withdraw from the study."
+* not clear on pre-registration next steps.'
+
+**Typos**
+
+* reported she "did not receive the sample".
+* chi-square would not be robust'. 
+
+The pattern we're looking for is "single or double quotation followed by a period and the end of the string," or `['"]\.$`.  If you want to see this in action, you can [try it on Regex 101](https://regex101.com/r/9eMUve/1).
+
+### Combining It All
+
+Let's combine start-of-string, except-for, and end-of-string, to match strings in our text which:
+
+* Have a start-of-string followed by a capital letter A-Z, then
+* Have twenty to thirty characters that cannot include digits and then
+* End in a final period (.) and end-of-string.
+
+We could do that with this expression: `^[A-Z][^\d]{10,30}\.$`. 
+
+* `^[A-Z]`: Start of string, capital letter.
+* `[^\d]{10,30}`: Anything that isn't a digit, 10 to 30 of them.
+* `\.$`: Literal period and end of string.
+
+[See what that looks like on Regex 101](https://regex101.com/r/9tHqoi/1) and in the image below:
+
+![A regular expression,`^[A-Z][^\d]{10,30}\.$`, is in the box labeled "Regular Expression," and there are five test strings typed in the "Test String" box. The test string that reads "This is a good string.", with a capital T at the beginning, is selected as a pattern match.  The second string, "this, not so good.", which begins with a lower case t, is not selected.  The third string reads "I'm an okay string." It is selected as a match.  The fourth string is "um, 3 things wrong here!", with a lower case starting letter, a numeral, and ending with an exclamation point. It is not selected. The final string says "I'm. A. Good. String."  Each word is capitalized and followed by a period.  It is marked as a match.](media/combining_it_all.png)<!-- style = "max-width: 300px; border: 1px solid rgb(var(--color-highlight));" -->
+
+### Scenario: Start- and End-of-String Anchors
+
+Consider these test strings, which are movie sequel titles:
+
+* Speed 2: Cruise Control
+* Teenage Mutant Ninja Turtles II: The Secret Of The Ooze
+* Tomb Raider: The Cradle Of Life
+
+You're helping a colleague who tried to write regex that would capture two groups for each movie: first, the part of the title before the colon, and second, the part of the title after the colon. 
+
+They used this regular expression, which uses capturing groups:  `([A-Za-z0-9\s]+):([A-Za-z0-9\s]+)`. [As we can see in Regex 101](https://regex101.com/r/3kyzIp/1), this did not segment the movie titles exactly right.  The regular expression matches the pattern twice, not three times.  It gets the first capturing group correct.  But for the second capturing group, the parser takes the second part of the first movie and keeps reading well into the second movie line, stopping only when it gets to a colon. 
```

Comment:
Question about this example: the screenshot shows that the `g` and `m` flags are on, which initially made me think that Regex101 would treat each line as its own separate string, so I was confused by this behavior. But looking more closely at the Regex101 documentation I see that the `m` flag actually means that `^` and `$` match the beginning and ends of lines, not that each line is its own string.  This leads me to things:

1. Correct the phrasing describing the `m` flag on the flags page
2. Move the flags section here so that you can then give one more example with anchors: `^([A-Za-z0-9\s]+):([A-Za-z0-9\s]+)$`

Thoughts?

Reply:
@drelliche actually I might leave flags where it belongs and just remove this scenario portion altogether, it might be confusing and un-helpful to people.  What do you think?

Reply:
I agree, let's take this scenario out entirely and save it for a future regex practice module.